src/installLogsCollector.js

```diff
@@ -90,24 +89,30 @@ function installLogsCollector(config = {}) {
   });
 
   afterEach(function() {
-    if (this.currentTest.state !== 'passed' || (config && config.printLogs === 'always')) {
-      // Need to wait otherwise some last commands get omitted from logs.
-      cy.wait(3, {log: false});
-      cy.task(CONSTANTS.TASK_NAME, {
-        spec: this.test.file,
-        test: this.currentTest.title,
-        messages: logs
-      }, {log: false});
-    }
+    // Need to wait otherwise some last commands get omitted from logs.
+    cy.wait(3, {log: false});
+    cy.task(CONSTANTS.TASK_NAME, {
+      spec: this.test.file,
+      test: this.currentTest.title,
+      messages: logs,
+      state: this.currentTest.state
+    }, {log: false});
   });
 
   after(function () {
     // Need to wait otherwise some last commands get omitted from logs.
-    cy.task(CONSTANTS.TASK_NAME_OUTPUT, null, {log: false});
+    cy.task(CONSTANTS.TASK_NAME_OUTPUT, {state: this.currentTest.state}, {log: false});
```

Comment:
This is incorrect. We need to know for each test from the spec whether they failed or not. The files should only contain the failing tests information in case the printLogsToFile is set to onFail.

Reply:
I don't know how to do this. While in one spec (default: state = "passed")
```
if (!(state === "passed" && this.currentTest.state === "passed")){
      state = this.currentTest.state;
    }
```
But I don't know how to determine when a spec test file has finished, so I can reset state to "passed".

Reply:
Actually that code is wrong.